src/box/iproto.cc

```diff
@@ -3041,36 +3090,92 @@ iproto_free(void)
 	evio_service_stop(&tx_binary);
 }
 
-int
-iproto_rmean_foreach(void *cb, void *cb_ctx)
+static void
+iproto_thread_rmean(struct iproto_thread *iproto_thread,
+		    int64_t *mean, int64_t *total, size_t name)
+{
+	assert(mean != NULL && total != NULL);
+	*mean += rmean_mean(iproto_thread->rmean, name);
+	*total += rmean_total(iproto_thread->rmean, name);
+}
+
+static void
+iproto_thread_tx_rmean(struct iproto_thread *iproto_thread,
+		       int64_t *mean, int64_t *total, size_t name)
+{
+	assert(mean != NULL && total != NULL);
+	*mean += rmean_mean(iproto_thread->tx.rmean, name);
+	*total += rmean_total(iproto_thread->tx.rmean, name);
+}
+
+typedef void (*rmean_stats_cb)(struct iproto_thread *iproto_thread,
+			       int64_t *mean, int64_t *total, size_t name);
+
+static int
+iproto_thread_rmean_foreach(const char **strings, rmean_stats_cb stats_cb,
+			    int thread_id, size_t rmean_last_name,
+			    void *cb, void *cb_ctx)
 {
-	for (size_t i = 0; i < RMEAN_NET_LAST; i++) {
+	int rc = 0;
+	for (size_t i = 0; i < rmean_last_name; i++) {
 		int64_t mean = 0;
 		int64_t total = 0;
-		for (int j = 0; j < iproto_threads_count; j++)  {
-			mean += rmean_mean(iproto_threads[j].rmean, i);
-			total += rmean_total(iproto_threads[j].rmean, i);
-		}
-		int rc = ((rmean_cb)cb)(rmean_net_strings[i], mean,
-					total, cb_ctx);
+		stats_cb(&iproto_threads[thread_id], &mean, &total, i);
```

Comment:
You don't need a new callback to accumulate stats. You can pass rmean instead. Please rework.

Reply:
Fixed